src/openms/include/OpenMS/MATH/STATISTICS/CumulativeHistogram.h

```diff
@@ -0,0 +1,179 @@
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2016.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Lukas Zimmermann $
+// $Authors: Lukas Zimmermann $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_MATH_STATISTICS_CUMULATIVEHISTOGRAM_H
+#define OPENMS_MATH_STATISTICS_CUMULATIVEHISTOGRAM_H
+
+//OpenMS
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CONCEPT/Exception.h>
+#include <OpenMS/MATH/STATISTICS/Histogram.h>
+
+//STL
+#include <vector>
+#include <cmath>
+#include <limits>
+#include <iostream>
+#include <algorithm>
+
+namespace OpenMS
+{
+  namespace Math
+  {
+    /**
+        @brief Representation of a histogram, which counts the cumulative number of observations.
+        This can be either in normal or complementary mode.
+
+        The first template argument gives the Type of the
+        values that are stored in the bins. The second argument
+        gives the type for the bin size and range.
+
+        @ingroup Math
+    */
+    template <typename ValueType = UInt, typename BinSizeType = double>
```

Comment:
I would actually just add a member getCummulativeHistogram to the orignal histogram class and maybe a member to increase all bin up to a value.